[BUG] nexusIQScan for Android not support multiple flavors

**Describe the bug**
On android we can declare additional "buildTypes" and "productFlavors". Unfortunately scanning does not work when project contain more than one flavor. Gradle raise exception:

> Caused by: org.gradle.api.GradleException: Could not scan the project: Cannot change dependencies of dependency configuration ':app:sonatypeCopyConfiguration' after it has been resolved.

**To Reproduce**
Steps to reproduce the behavior:
1. Create android project with multiple flavors
2. Connect android library as submodule
3. Run simulated scanning eg. `gradle --stacktrace nexusIQScan`

Sample project containing this error: https://github.com/Shusek/SonarGradlePluginFlavorBug
**Expected behavior**
Plugin should support more than one flavor

**Desktop (please complete the following information):**
 - OS: Linux Ubuntu
 - Gradle Version: 6.7.1
 - JVM Version and Flavor: Android Studio embedded JVM 1.8
 - Nexus IQ Server Version: dry run
 - Version: 2.0.0

**Additional context**
Full stacktrace error:

> `* What went wrong:
> Execution failed for task ':nexusIQScan'.
> > Could not scan the project: Cannot change dependencies of dependency configuration ':app:sonatypeCopyConfiguration' after it has been resolved.
> 
> * Try:
> Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> 
> * Exception is:
> org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':nexusIQScan'.
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:207)
>         at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:263)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:205)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:186)
>         at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:114)
>         at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
>         at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
>         at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
>         at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
>         at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
>         at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
>         at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
>         at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:409)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:399)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:94)
>         at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
>         at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
>         at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
>         at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:356)
>         at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:343)
>         at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:336)
>         at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:322)
>         at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
>         at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
>         at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
>         at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
>         at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
>         at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
>         at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
> Caused by: org.gradle.api.GradleException: Could not scan the project: Cannot change dependencies of dependency configuration ':app:sonatypeCopyConfiguration' after it has been resolved.
>         at org.sonatype.gradle.plugins.scan.nexus.iq.NexusIqScanTask.scan(NexusIqScanTask.java:118)
>         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>         at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
>         at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:49)
>         at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:42)
>         at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:28)
>         at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:726)
>         at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:693)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.run(ExecuteActionsTaskExecuter.java:569)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:395)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:387)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:84)
>         at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:554)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:537)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.access$300(ExecuteActionsTaskExecuter.java:108)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.executeWithPreviousOutputFiles(ExecuteActionsTaskExecuter.java:278)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.execute(ExecuteActionsTaskExecuter.java:267)
>         at org.gradle.internal.execution.steps.ExecuteStep.lambda$execute$1(ExecuteStep.java:33)
>         at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:33)
>         at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:26)
>         at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:67)
>         at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:36)
>         at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:49)
>         at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:34)
>         at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:43)
>         at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:73)
>         at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:54)
>         at org.gradle.internal.execution.steps.CatchExceptionStep.execute(CatchExceptionStep.java:34)
>         at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:44)
>         at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:54)
>         at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:38)
>         at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:49)
>         at org.gradle.internal.execution.steps.CacheStep.executeWithoutCache(CacheStep.java:159)
>         at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:72)
>         at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:43)
>         at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:44)
>         at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:33)
>         at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:38)
>         at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:24)
>         at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:92)
>         at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$0(SkipUpToDateStep.java:85)
>         at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:55)
>         at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:39)
>         at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:76)
>         at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:37)
>         at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:36)
>         at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:26)
>         at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:94)
>         at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:49)
>         at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:79)
>         at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:53)
>         at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:74)
>         at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:78)
>         at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:78)
>         at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:34)
>         at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:39)
>         at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:40)
>         at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:28)
>         at org.gradle.internal.execution.impl.DefaultWorkExecutor.execute(DefaultWorkExecutor.java:33)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:194)
>         at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:186)
>         at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:114)
>         at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
>         at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
>         at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
>         at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
>         at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
>         at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
>         at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
>         at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:409)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:399)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:94)
>         at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
>         at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
>         at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
>         at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:356)
>         at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:343)
>         at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:336)
>         at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:322)
>         at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
>         at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
>         at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
>         at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
>         at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
>         at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
>         at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
> Caused by: org.gradle.api.InvalidUserDataException: Cannot change dependencies of dependency configuration ':app:sonatypeCopyConfiguration' after it has been resolved.
>         at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.preventIllegalMutation(DefaultConfiguration.java:1128)
>         at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.validateMutation(DefaultConfiguration.java:1099)
>         at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.lambda$validateMutationType$1(DefaultConfiguration.java:298)
>         at org.gradle.internal.ImmutableActionSet$SingletonSet.execute(ImmutableActionSet.java:225)
>         at org.gradle.api.internal.DefaultDomainObjectSet.assertMutableCollectionContents(DefaultDomainObjectSet.java:71)
>         at org.gradle.api.internal.DefaultDomainObjectCollection.add(DefaultDomainObjectCollection.java:252)
>         at org.gradle.api.internal.DelegatingDomainObjectSet.add(DelegatingDomainObjectSet.java:110)
>         at org.gradle.api.internal.artifacts.DefaultDependencySet.add(DefaultDependencySet.java:68)
>         at org.gradle.api.internal.artifacts.DefaultDependencySet.add(DefaultDependencySet.java:36)
>         at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.lambda$getDependencies$10(DependenciesFinder.java:151)
>         at org.gradle.api.internal.CompositeDomainObjectSet.all(CompositeDomainObjectSet.java:158)
>         at org.gradle.api.internal.DelegatingDomainObjectSet.all(DelegatingDomainObjectSet.java:55)
>         at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.getDependencies(DependenciesFinder.java:145)
>         at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.lambda$findResolvedArtifacts$6(DependenciesFinder.java:76)
>         at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.findResolvedArtifacts(DependenciesFinder.java:78)
>         at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.lambda$findModules$8(DependenciesFinder.java:87)
>         at org.gradle.api.internal.DefaultMutationGuard$2.execute(DefaultMutationGuard.java:42)
>         at org.gradle.internal.Actions.with(Actions.java:251)
>         at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator$2$1.run(BuildOperationCrossProjectConfigurator.java:79)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:395)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:387)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:84)
>         at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
>         at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator$2.run(BuildOperationCrossProjectConfigurator.java:76)
>         at org.gradle.internal.Factories$1.create(Factories.java:26)
>         at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withMutableState(DefaultProjectStateRegistry.java:245)
>         at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withMutableState(DefaultProjectStateRegistry.java:226)
>         at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator.runProjectConfigureAction(BuildOperationCrossProjectConfigurator.java:73)
>         at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator.access$400(BuildOperationCrossProjectConfigurator.java:32)
>         at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator$1.doRunProjectConfigure(BuildOperationCrossProjectConfigurator.java:67)
>         at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator$BlockConfigureBuildOperation.run(BuildOperationCrossProjectConfigurator.java:121)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:395)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:387)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
>         at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:84)
>         at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
>         at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator.runBlockConfigureAction(BuildOperationCrossProjectConfigurator.java:64)
>         at org.gradle.api.internal.project.BuildOperationCrossProjectConfigurator.allprojects(BuildOperationCrossProjectConfigurator.java:54)
>         at org.gradle.api.internal.project.DefaultProject.allprojects(DefaultProject.java:676)
>         at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.findModules(DependenciesFinder.java:84)
>         at org.sonatype.gradle.plugins.scan.nexus.iq.NexusIqScanTask.scan(NexusIqScanTask.java:78)
> `
>

--- a/src/integTest/java/org/sonatype/gradle/plugins/scan/ScanPluginIntegrationTestBase.java
+++ b/src/integTest/java/org/sonatype/gradle/plugins/scan/ScanPluginIntegrationTestBase.java
@@ -149,6 +149,23 @@ public abstract class ScanPluginIntegrationTestBase
   }
 
   @Test
+  public void testScanTask_AndroidMultipleFlavors_NexusIQ() {
+    String resource =
+        useLegacySyntax ? "legacy-syntax" + File.separator + "android-multiple-flavors" : "android-multiple-flavors";
+    ResourcesUtils.copyFromClassPath(resource, testProjectDir.getRoot().toPath());
+
+    final BuildResult result = GradleRunner.create()
+        .withGradleVersion(gradleVersion)
+        .withProjectDir(new File(testProjectDir.getRoot(), resource))
+        .withPluginClasspath()
+        .withArguments("nexusIQScan", "--info")
+        .build();
+
+    assertBuildOutputText_NexusIQ(result, "None");
+    assertThat(result.task(":nexusIQScan").getOutcome()).isEqualTo(SUCCESS);
+  }
+
+  @Test
   public void testAuditTask_NoVulnerabilities_OssIndex_Default_Empty() throws IOException {
     writeFile(buildFile, "control_default_not_all.gradle");
 
@@ -325,6 +342,30 @@ public abstract class ScanPluginIntegrationTestBase
   @Test
   public void testAuditTask_Android_OssIndex() {
     String resource = useLegacySyntax ? "legacy-syntax" + File.separator + "android" : "android";
+    ResourcesUtils.copyFromClassPath(resource, testProjectDir.getRoot().toPath());
+
+    BuildResult result = GradleRunner.create()
+        .withGradleVersion(gradleVersion)
+        .withProjectDir(new File(testProjectDir.getRoot(), resource))
+        .withPluginClasspath()
+        .withArguments("ossIndexAudit", "--info")
+        .build();
+
+    String resultOutput = result.getOutput();
+    assertThat(resultOutput).contains(String.format(
+        "%scom.android.support:appcompat-v7:24.2.1: 0 vulnerabilities detected%n"
+            + "|    %scom.android.support:animated-vector-drawable:24.2.1: 0 vulnerabilities detected",
+        DEPENDENCY_PREFIX, DEPENDENCY_PREFIX));
+    assertThat(resultOutput).contains(
+        String.format("%scommons-collections:commons-collections:3.1: 0 vulnerabilities detected", DEPENDENCY_PREFIX));
+
+    assertThat(result.task(":ossIndexAudit").getOutcome()).isEqualTo(SUCCESS);
+  }
+
+  @Test
+  public void testAuditTask_AndroidMultipleFlavors_OssIndex() {
+    String resource =
+        useLegacySyntax ? "legacy-syntax" + File.separator + "android-multiple-flavors" : "android-multiple-flavors";
     ResourcesUtils.copyFromClassPath(resource, testProjectDir.getRoot().toPath());
 
     BuildResult result = GradleRunner.create()

--- a/src/integTest/resources/android-multiple-flavors/app/build.gradle
+++ b/src/integTest/resources/android-multiple-flavors/app/build.gradle
@@ -1,0 +1,34 @@
+apply plugin: 'com.android.application'
+
+android {
+  compileSdkVersion rootProject.ext.compileSdkVersion
+
+  defaultConfig {
+    applicationId 'org.sonatype.gradle.plugins.scan.android'
+    minSdkVersion rootProject.ext.minSdkVersion
+    targetSdkVersion rootProject.ext.targetSdkVersion
+    versionCode 1
+    versionName '1.0.0'
+    buildToolsVersion rootProject.ext.buildToolsVersion
+  }
+
+  flavorDimensions 'version'
+
+  productFlavors {
+    demo {
+      dimension 'version'
+      applicationIdSuffix '.flavor1'
+      versionNameSuffix '-flavor1'
+    }
+    full {
+      dimension 'version'
+      applicationIdSuffix '.flavor2'
+      versionNameSuffix '-flavor2'
+    }
+  }
+}
+
+dependencies {
+  implementation 'com.android.support:appcompat-v7:24.2.1'
+  implementation project(path: ':common-lib')
+}

--- a/src/integTest/resources/android-multiple-flavors/app/src/main/AndroidManifest.xml
+++ b/src/integTest/resources/android-multiple-flavors/app/src/main/AndroidManifest.xml
@@ -1,0 +1,5 @@
+<manifest package="com.inventage.sonatype.android.app"
+          xmlns:android="http://schemas.android.com/apk/res/android"
+          xmlns:tools="http://schemas.android.com/tools">
+  <application android:name="org.sonatype.gradle.plugins.scan.android" />
+</manifest>

--- a/src/integTest/resources/android-multiple-flavors/build.gradle
+++ b/src/integTest/resources/android-multiple-flavors/build.gradle
@@ -1,0 +1,53 @@
+buildscript {
+  repositories {
+    google()
+    mavenCentral()
+    jcenter()
+  }
+
+  // Mapping between Gradle and Android plugin versions: https://developer.android.com/studio/releases/gradle-plugin
+  def androidGradlePluginVersion = '3.0.0'
+  if (gradle.gradleVersion.compareTo('6.1.1') >= 0) {
+    androidGradlePluginVersion = '4.0.0'
+  } else if (gradle.gradleVersion.compareTo('5.6.4') >= 0) {
+    androidGradlePluginVersion = '3.6.4'
+  } else if (gradle.gradleVersion.compareTo('5.0') >= 0) {
+    androidGradlePluginVersion = '3.3.3'
+  }
+  dependencies {
+    classpath "com.android.tools.build:gradle:$androidGradlePluginVersion"
+  }
+}
+
+plugins {
+  id 'org.sonatype.gradle.plugins.scan'
+}
+
+allprojects {
+  repositories {
+    google()
+    mavenCentral()
+    jcenter()
+  }
+}
+
+ext {
+  compileSdkVersion=28
+  minSdkVersion=10
+  targetSdkVersion=28
+  buildToolsVersion='26.0.2'
+}
+
+nexusIQScan {
+  serverUrl = 'http://localhost:8081'
+  username = 'admin'
+  password = 'admin123'
+  applicationId = 'testing-gradle-plugin'
+  simulationEnabled = true
+}
+
+ossIndexAudit {
+  simulationEnabled = true
+  dependencyGraph = true
+  showAll = true
+}

--- a/src/integTest/resources/android-multiple-flavors/build.gradle
+++ b/src/integTest/resources/android-multiple-flavors/build.gradle
@@ -14,6 +14,7 @@ buildscript {
   } else if (gradle.gradleVersion.compareTo('5.0') >= 0) {
     androidGradlePluginVersion = '3.3.3'
   }
+
   dependencies {
     classpath "com.android.tools.build:gradle:$androidGradlePluginVersion"
   }

--- a/src/integTest/resources/android-multiple-flavors/common-lib/build.gradle
+++ b/src/integTest/resources/android-multiple-flavors/common-lib/build.gradle
@@ -1,0 +1,17 @@
+apply plugin: 'com.android.library'
+
+android {
+  compileSdkVersion rootProject.ext.compileSdkVersion
+
+  defaultConfig {
+    minSdkVersion rootProject.ext.minSdkVersion
+    targetSdkVersion rootProject.ext.targetSdkVersion
+    versionCode 1
+    versionName '1.0'
+    buildToolsVersion rootProject.ext.buildToolsVersion
+  }
+}
+
+dependencies {
+  implementation 'commons-collections:commons-collections:3.1'
+}

--- a/src/integTest/resources/android-multiple-flavors/common-lib/src/main/AndroidManifest.xml
+++ b/src/integTest/resources/android-multiple-flavors/common-lib/src/main/AndroidManifest.xml
@@ -1,0 +1,5 @@
+<manifest package="com.inventage.sonatype.android.app"
+          xmlns:android="http://schemas.android.com/apk/res/android"
+          xmlns:tools="http://schemas.android.com/tools">
+  <application android:name="org.sonatype.gradle.plugins.scan.android" />
+</manifest>

--- a/src/integTest/resources/android-multiple-flavors/settings.gradle
+++ b/src/integTest/resources/android-multiple-flavors/settings.gradle
@@ -1,0 +1,1 @@
+include ':common-lib', ':app'

--- a/src/integTest/resources/legacy-syntax/android-multiple-flavors/app/build.gradle
+++ b/src/integTest/resources/legacy-syntax/android-multiple-flavors/app/build.gradle
@@ -1,0 +1,34 @@
+apply plugin: 'com.android.application'
+
+android {
+  compileSdkVersion rootProject.ext.compileSdkVersion
+
+  defaultConfig {
+    applicationId 'org.sonatype.gradle.plugins.scan.android'
+    minSdkVersion rootProject.ext.minSdkVersion
+    targetSdkVersion rootProject.ext.targetSdkVersion
+    versionCode 1
+    versionName '1.0.0'
+    buildToolsVersion rootProject.ext.buildToolsVersion
+  }
+
+  flavorDimensions 'version'
+
+  productFlavors {
+    flavor1 {
+      dimension 'version'
+      applicationIdSuffix '.flavor1'
+      versionNameSuffix '-flavor1'
+    }
+    flavor2 {
+      dimension 'version'
+      applicationIdSuffix '.flavor2'
+      versionNameSuffix '-flavor2'
+    }
+  }
+}
+
+dependencies {
+  compile 'com.android.support:appcompat-v7:24.2.1'
+  compile project(path: ':common-lib')
+}

--- a/src/integTest/resources/legacy-syntax/android-multiple-flavors/app/src/main/AndroidManifest.xml
+++ b/src/integTest/resources/legacy-syntax/android-multiple-flavors/app/src/main/AndroidManifest.xml
@@ -1,0 +1,5 @@
+<manifest package="com.inventage.sonatype.android.app"
+          xmlns:android="http://schemas.android.com/apk/res/android"
+          xmlns:tools="http://schemas.android.com/tools">
+  <application android:name="org.sonatype.gradle.plugins.scan.android" />
+</manifest>

--- a/src/integTest/resources/legacy-syntax/android-multiple-flavors/build.gradle
+++ b/src/integTest/resources/legacy-syntax/android-multiple-flavors/build.gradle
@@ -1,0 +1,43 @@
+buildscript {
+  repositories {
+    maven { url 'https://maven.google.com' }
+    mavenCentral()
+  }
+
+  dependencies {
+    classpath 'com.android.tools.build:gradle:2.3.0'
+  }
+}
+
+plugins {
+  id 'org.sonatype.gradle.plugins.scan'
+}
+
+allprojects {
+  repositories {
+    maven { url 'https://maven.google.com' }
+    mavenCentral()
+  }
+}
+
+ext {
+  compileSdkVersion=28
+  minSdkVersion=10
+  targetSdkVersion=28
+  buildToolsVersion='26.0.2'
+}
+
+nexusIQScan {
+  serverUrl = 'http://localhost:8081'
+  username = 'admin'
+  password = 'admin123'
+  applicationId = 'testing-gradle-plugin'
+  simulationEnabled = true
+}
+
+ossIndexAudit {
+  simulationEnabled = true
+  dependencyGraph = true
+  showAll = true
+  allConfigurations = true
+}

--- a/src/integTest/resources/legacy-syntax/android-multiple-flavors/common-lib/build.gradle
+++ b/src/integTest/resources/legacy-syntax/android-multiple-flavors/common-lib/build.gradle
@@ -1,0 +1,17 @@
+apply plugin: 'com.android.library'
+
+android {
+  compileSdkVersion rootProject.ext.compileSdkVersion
+
+  defaultConfig {
+    minSdkVersion rootProject.ext.minSdkVersion
+    targetSdkVersion rootProject.ext.targetSdkVersion
+    versionCode 1
+    versionName '1.0'
+    buildToolsVersion rootProject.ext.buildToolsVersion
+  }
+}
+
+dependencies {
+  compile 'commons-collections:commons-collections:3.1'
+}

--- a/src/integTest/resources/legacy-syntax/android-multiple-flavors/common-lib/src/main/AndroidManifest.xml
+++ b/src/integTest/resources/legacy-syntax/android-multiple-flavors/common-lib/src/main/AndroidManifest.xml
@@ -1,0 +1,5 @@
+<manifest package="com.inventage.sonatype.android.app"
+          xmlns:android="http://schemas.android.com/apk/res/android"
+          xmlns:tools="http://schemas.android.com/tools">
+  <application android:name="org.sonatype.gradle.plugins.scan.android" />
+</manifest>

--- a/src/integTest/resources/legacy-syntax/android-multiple-flavors/settings.gradle
+++ b/src/integTest/resources/legacy-syntax/android-multiple-flavors/settings.gradle
@@ -1,0 +1,1 @@
+include ':common-lib', ':app'

--- a/src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java
+++ b/src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java
@@ -59,9 +59,6 @@ public class DependenciesFinder
   private static final String ATTRIBUTES_SUPPORTED_GRADLE_VERSION = "4.0";
 
   public Set<ResolvedDependency> findResolvedDependencies(Project rootProject, boolean allConfigurations) {
-    rootProject.getAllprojects().forEach(p -> p.getConfigurations().forEach(c -> {
-      System.out.println(c.getAllArtifacts().size() + ": " + c.getName());
-    }));
     return new LinkedHashSet<>(rootProject.getAllprojects()).stream()
         .flatMap(project -> project.getConfigurations().stream())
         .filter(configuration -> isAcceptableConfiguration(configuration, allConfigurations))
@@ -117,9 +114,8 @@ public class DependenciesFinder
   @VisibleForTesting
   Configuration createCopyConfiguration(Project project) {
     String configurationName = COPY_CONFIGURATION_NAME;
-    int i = 0;
-    while (project.getConfigurations().findByName(configurationName) != null) {
-      configurationName += ++i;
+    for (int i = 0; project.getConfigurations().findByName(configurationName) != null; i++) {
+      configurationName += i;
     }
     Configuration copyConfiguration = project.getConfigurations().create(configurationName);
     if (isGradleVersionSupportedForAttributes(project.getGradle().getGradleVersion())) {

--- a/src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java
+++ b/src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java
@@ -59,21 +59,21 @@ public class DependenciesFinder
   private static final String ATTRIBUTES_SUPPORTED_GRADLE_VERSION = "4.0";
 
   public Set<ResolvedDependency> findResolvedDependencies(Project rootProject, boolean allConfigurations) {
-    Configuration copyConfiguration = createCopyConfiguration(rootProject);
-
-    return rootProject.getAllprojects().stream().flatMap(project -> project.getConfigurations().stream())
+    rootProject.getAllprojects().forEach(p -> p.getConfigurations().forEach(c -> {
+      System.out.println(c.getAllArtifacts().size() + ": " + c.getName());
+    }));
+    return new LinkedHashSet<>(rootProject.getAllprojects()).stream()
+        .flatMap(project -> project.getConfigurations().stream())
         .filter(configuration -> isAcceptableConfiguration(configuration, allConfigurations))
-        .flatMap(configuration -> getDependencies(rootProject, configuration, copyConfiguration,
+        .flatMap(configuration -> getDependencies(rootProject, configuration,
             resolvedConfiguration -> resolvedConfiguration.getFirstLevelModuleDependencies().stream()))
         .collect(Collectors.toCollection(LinkedHashSet::new));
   }
 
   public Set<ResolvedArtifact> findResolvedArtifacts(Project project, boolean allConfigurations) {
-    Configuration copyConfiguration = createCopyConfiguration(project);
-
-    return project.getConfigurations().stream()
+    return new LinkedHashSet<>(project.getConfigurations()).stream()
         .filter(configuration -> isAcceptableConfiguration(configuration, allConfigurations))
-        .flatMap(configuration -> getDependencies(project, configuration, copyConfiguration,
+        .flatMap(configuration -> getDependencies(project, configuration,
             resolvedConfiguration -> resolvedConfiguration.getResolvedArtifacts().stream()))
         .collect(Collectors.toSet());
   }
@@ -116,7 +116,12 @@ public class DependenciesFinder
 
   @VisibleForTesting
   Configuration createCopyConfiguration(Project project) {
-    Configuration copyConfiguration = project.getConfigurations().maybeCreate(COPY_CONFIGURATION_NAME);
+    String configurationName = COPY_CONFIGURATION_NAME;
+    int i = 0;
+    while (project.getConfigurations().findByName(configurationName) != null) {
+      configurationName += ++i;
+    }
+    Configuration copyConfiguration = project.getConfigurations().create(configurationName);
     if (isGradleVersionSupportedForAttributes(project.getGradle().getGradleVersion())) {
       copyConfiguration.attributes(attributeContainer -> {
         ObjectFactory factory = project.getObjects();
@@ -135,13 +140,14 @@ public class DependenciesFinder
   <T> Stream<T> getDependencies(
       Project project,
       Configuration originalConfiguration,
-      Configuration copyConfiguration,
       Function<ResolvedConfiguration, Stream<T>> function)
   {
     try {
       return function.apply(originalConfiguration.getResolvedConfiguration());
     }
     catch (ResolveException e) {
+      Configuration copyConfiguration = createCopyConfiguration(project);
+
       originalConfiguration.getAllDependencies().all(dependency -> {
         if (dependency instanceof ProjectDependency) {
           Project dependencyProject = ((ProjectDependency) dependency).getDependencyProject();
@@ -151,6 +157,7 @@ public class DependenciesFinder
           copyConfiguration.getDependencies().add(dependency);
         }
       });
+
       return function.apply(copyConfiguration.getResolvedConfiguration());
     }
   }

--- a/src/test/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinderTest.java
+++ b/src/test/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinderTest.java
@@ -221,9 +221,8 @@ public class DependenciesFinderTest
   public void testGetDependencies_ModuleDependencies() {
     Project project = buildProject(COMPILE_CLASSPATH_CONFIGURATION_NAME, false);
     Configuration originalConfiguration = project.getConfigurations().getByName(COMPILE_CLASSPATH_CONFIGURATION_NAME);
-    Configuration copyConfiguration = finder.createCopyConfiguration(project);
 
-    Stream<ResolvedDependency> dependencies = finder.getDependencies(project, originalConfiguration, copyConfiguration,
+    Stream<ResolvedDependency> dependencies = finder.getDependencies(project, originalConfiguration,
         resolvedConfiguration -> resolvedConfiguration.getFirstLevelModuleDependencies().stream());
 
     assertThat(dependencies).isNotNull();
@@ -246,9 +245,7 @@ public class DependenciesFinderTest
     when(originalConfiguration.getAllDependencies())
         .thenReturn(project.getConfigurations().getByName(COMPILE_CLASSPATH_CONFIGURATION_NAME).getAllDependencies());
 
-    Configuration copyConfiguration = finder.createCopyConfiguration(project);
-
-    Stream<ResolvedDependency> dependencies = finder.getDependencies(project, originalConfiguration, copyConfiguration,
+    Stream<ResolvedDependency> dependencies = finder.getDependencies(project, originalConfiguration,
         resolvedConfiguration -> resolvedConfiguration.getFirstLevelModuleDependencies().stream());
 
     assertThat(dependencies).isNotNull();
@@ -266,9 +263,8 @@ public class DependenciesFinderTest
   public void testGetDependencies_ResolvedArtifacts() {
     Project project = buildProject(COMPILE_CLASSPATH_CONFIGURATION_NAME, false);
     Configuration originalConfiguration = project.getConfigurations().getByName(COMPILE_CLASSPATH_CONFIGURATION_NAME);
-    Configuration copyConfiguration = finder.createCopyConfiguration(project);
 
-    Stream<ResolvedArtifact> dependencies = finder.getDependencies(project, originalConfiguration, copyConfiguration,
+    Stream<ResolvedArtifact> dependencies = finder.getDependencies(project, originalConfiguration,
         resolvedConfiguration -> resolvedConfiguration.getResolvedArtifacts().stream());
 
     assertThat(dependencies).isNotNull();
@@ -288,9 +284,7 @@ public class DependenciesFinderTest
     when(originalConfiguration.getAllDependencies())
         .thenReturn(project.getConfigurations().getByName(COMPILE_CLASSPATH_CONFIGURATION_NAME).getAllDependencies());
 
-    Configuration copyConfiguration = finder.createCopyConfiguration(project);
-
-    Stream<ResolvedArtifact> dependencies = finder.getDependencies(project, originalConfiguration, copyConfiguration,
+    Stream<ResolvedArtifact> dependencies = finder.getDependencies(project, originalConfiguration,
         resolvedConfiguration -> resolvedConfiguration.getResolvedArtifacts().stream());
 
     assertThat(dependencies).isNotNull();

